File: data/data-implementation/src/commonMain/kotlin/data/implementation/domain/source/KeystoreManagerImpl.kt
package data.implementation.domain.source

import data.api.domain.source.KeystoreManager
import data.api.model.security.KeyModel
import org.koin.core.annotation.Factory
import kotlin.uuid.ExperimentalUuidApi
import kotlin.uuid.Uuid

@Factory
class KeystoreManagerImpl : KeystoreManager {
    @OptIn(ExperimentalUuidApi::class)
    override fun generateEcKeyPair(): KeyModel {
        // TODO implement Android EC Key generation using Nimbus library
        return KeyModel(
            keyId = Uuid.random().toString(),
            privateKey = "",
            publicKey = ""
        )
    }
}

File: iosApp/iosApp/Domain/KeystoreManagerImpl.swift
import Foundation
import ComposeApp
import CryptoKit

class KeystoreManagerImpl : KeystoreManager {
    func generateEcKeyPair() -> KeyModel {
        let privateKey = P256.KeyAgreement.PrivateKey()
        let publicKey = privateKey.publicKey.derRepresentation.base64EncodedString()
        return KeyModel(keyId: UUID().uuidString, privateKey: privateKey.derRepresentation.base64EncodedString(), publicKey: publicKey)
    }
}
